xonsh/ptk/shell.py

```diff
@@ -142,8 +142,6 @@ def rprompt_tokens(self, cli):
         prompt.
         """
         p = builtins.__xonsh_env__.get('RIGHT_PROMPT')
-        if len(p) == 0:
-            return []
```

Comment:
This is as an optimization to avoid a whole, relatively expensive branch.  You are correct that this won't effect the result of the execution, but it will make the default case of not having a right prompt faster, which is what I was after originally.  I hope that this makes some sense :)